R/run-nmtran.R

```diff
@@ -0,0 +1,330 @@
+          run_dir = run_dir,
+          status = "nmtran_presort failed. See errors.",
+          status_val = presort_status_val,
+          output_lines = presort.p$stdout
+        )
+      )
+    }
+
+    # Write the output to tempzzzz1 control stream
+    writeLines(presort.p$stdout, con = file.path(run_dir, "tempzzzz1.ctl"))
+  }
+
+  # Run NM-TRAN
+  stdin_file <- ifelse(run_presort, "tempzzzz1.ctl", mod_path)
+  nmtran.p <- processx::run(
+    command = nmtran_exe, wd = run_dir, args = cmd_args,
+    stdout = "|", stdin = file.path(run_dir, stdin_file),
+    stderr_to_stdout = TRUE, error_on_status = FALSE
+  )
+
+  # Assign status
+  status_val <- nmtran.p$status
+  if(is.na(status_val)){
+    rlang::abort("NM-TRAN terminated unexpectedly")
+  }else if(status_val == 0){
+    status <- "NM-TRAN successful"
+  }else{
+    status <- "NM-TRAN failed. See errors."
+  }
+
+  # Tabulate NM-TRAN results
+  nmtran_results <- list(
+    nmtran_model = stdin_file,
+    run_dir = run_dir,
+    status = status,
+    status_val = status_val,
+    output_lines = nmtran.p$stdout
+  )
+
+  return(nmtran_results)
+}
+
+
+#' Helper for finding and reading in a `bbi.yaml` file given a `bbi_nonmem_model`
+#' object or explicit `.config_path`.
+#' @inheritParams run_nmtran
+#' @noRd
+get_bbi_yaml_path <- function(.mod = NULL, .config_path = NULL){
+  if(!is.null(.mod)){
+    check_model_object(.mod, "bbi_nonmem_model")
+    model_dir <- get_model_working_directory(.mod)
+  }
+  bbi_yaml_path <- .config_path %||% file.path(model_dir, "bbi.yaml")
+
+  if(!file_exists(bbi_yaml_path)){
+    if(is.null(.config_path)){
+      msg <- c(
+        "No bbi configuration was found in the execution directory.",
+        "i" = "Please run `bbi_init()` with the appropriate directory to continue."
+      )
+    }else{
+      msg <- glue("No bbi configuration was found at {.config_path}")
+    }
+    rlang::abort(msg)
+  }
+
+  return(bbi_yaml_path)
+}
+
+
+#' Parse `.bbi_args` and return the three expected `nmfe_options` for `NM-TRAN`
+#'  in the correct format
+#' @inheritParams run_nmtran
+#' @param nmfe_options named list of nmfe options defined in a `bbi.yaml` file.
+#' @noRd
+parse_nmtran_args <- function(
+    .mod,
+    .bbi_args = NULL,
+    nmfe_options = NULL
+){
+
+  nmfe_args_def <- list(prdefault = FALSE, tprdefault = FALSE, maxlim = 2)
+
+  # Combine with any options stored in model yaml, preferring .bbi_args
+  .nmfe_args <- parse_args_list(.bbi_args, .mod[[YAML_BBI_ARGS]])
+
+  # Combine with nmfe options stored in bbi.yaml, preferring .nmfe_args
+  .nmfe_args <- parse_args_list(.nmfe_args, nmfe_options)
+
+  # Check provided args
+  check_bbi_args(.nmfe_args)
+
+  # Combine with and filter to default nmfe_options
+  .nmfe_args <- parse_args_list(.nmfe_args, nmfe_args_def)
+  .nmfe_args <- .nmfe_args[names(nmfe_args_def)]
+
+  .nmtran_args <- c(
+    ifelse(isTRUE(.nmfe_args$prdefault), 1, 0),
+    ifelse(isTRUE(.nmfe_args$tprdefault), 1, 0),
```

Comment:
These are operating where the input and output should always be a scalar, so please don't use the vectorized `ifelse`.